### ISSUE TITLE
[cgal] build failure with a manifest

### ISSUE THREAD

**Issue by LegalizeAdulthood** [microsoft/vcpkg#16687]
**Host Environment**
- Windows 10
- Compiler: VS 2019

**To Reproduce**
With the following manifest file:
```
{
    "name": "goink",
    "version-string": "1.0.0",
    "dependencies": [
        "cgal"
    ]
}
```
and the following CMakeLists.txt in a source directory:
```
cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
project(goink CXX)
```
Then configure a CMake build using vcpkg as the toolchain:
`cmake -DCMAKE_TOOLCHAIN_FILE:FILEPATH=<path to vcpkg>/scripts/buildsystems/vcpkg.cmake <path to source>`
When vcpkg attempts to build the gmp dependency of cgal, it fails.

**Failure logs** 
```
Starting package 4/89: gmp:x64-windows
Building package gmp[core]:x64-windows...
-- Using cached D:/vcpkg/downloads/ShiftMediaProject-gmp-0018c44e8dfcc3b64b43e0aea4b3f419f0b65fd0.tar.gz
-- Cleaning sources at D:/vcpkg/buildtrees/gmp/src/19f0b65fd0-9b395bc164.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/vcpkg/downloads/ShiftMediaProject-gmp-0018c44e8dfcc3b64b43e0aea4b3f419f0b65fd0.tar.gz
-- Applying patch vs.build.patch
-- Applying patch runtime.patch
-- Applying patch adddef.patch
-- Using source at D:/vcpkg/buildtrees/gmp/src/19f0b65fd0-9b395bc164.clean
CMake Error at D:/legalize/cgal/goink-build/vcpkg_installed/x64-windows/share/yasm-tool-helper/yasm-tool-helper.cmake:16 (message):
  Cross-targetting and x64 ports requiring yasm require the x86-windows
  yasm-tool to be available.  Please install yasm-tool:x86-windows first.
Call Stack (most recent call first):
  ports/gmp/portfile.cmake:19 (yasm_tool_helper)
  scripts/ports.cmake:142 (include)


Error: Building package gmp:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: gmp:x64-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

Additionally, attach any relevant sections from the log files above.
-- Running vcpkg install - failed
CMake Error at D:/vcpkg/scripts/buildsystems/vcpkg.cmake:898 (message):
  vcpkg install failed.  See logs for more information:
  D:\legalize\cgal\goink-build\vcpkg-manifest-install.log
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.19/Modules/CMakeDetermineSystem.cmake:123 (include)
  CMakeLists.txt:3 (project)


-- Configuring incomplete, errors occurred!
```

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

**Comment by LegalizeAdulthood** [comment]
Attempting `vcpkg install cgal` seems to be able to build gmp without errors:
```
Starting package 87/90: gmp:x86-windows
Building package gmp[core]:x86-windows...
-- Using cached D:/vcpkg/downloads/ShiftMediaProject-gmp-0018c44e8dfcc3b64b43e0aea4b3f419f0b65fd0.tar.gz
-- Cleaning sources at D:/vcpkg/buildtrees/gmp/src/19f0b65fd0-9b395bc164.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/vcpkg/downloads/ShiftMediaProject-gmp-0018c44e8dfcc3b64b43e0aea4b3f419f0b65fd0.tar.gz
-- Applying patch vs.build.patch
-- Applying patch runtime.patch
-- Applying patch adddef.patch
-- Using source at D:/vcpkg/buildtrees/gmp/src/19f0b65fd0-9b395bc164.clean
-- Building SMP/libgmp.sln for Release
-- Building SMP/libgmp.sln for Debug
-- Installing: D:/vcpkg/packages/gmp_x86-windows/share/gmp/copyright
-- Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-2-any.pkg.tar.zst -> msys-mingw-w64-i686-pkg-config-0.29.2-2-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst -> msys-mingw-w64-i686-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst...
-- Using msys root at D:/vcpkg/downloads/tools/msys2/50864d9fae363d6e
-- Fixing pkgconfig file: D:/vcpkg/packages/gmp_x86-windows/lib/pkgconfig/gmp.pc
-- Fixing pkgconfig file: D:/vcpkg/packages/gmp_x86-windows/lib/pkgconfig/gmpxx.pc
-- Fixing pkgconfig file: D:/vcpkg/packages/gmp_x86-windows/debug/lib/pkgconfig/gmp.pc
-- Fixing pkgconfig file: D:/vcpkg/packages/gmp_x86-windows/debug/lib/pkgconfig/gmpxx.pc
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\rthomson.NVIDIA.COM\AppData\Local\vcpkg\archives\9d\9d0f5d3d419b3ba82569444346c4b3def3d8c061.zip
Building package gmp[core]:x86-windows... done
Installing package gmp[core]:x86-windows...
Installing package gmp[core]:x86-windows... done
Elapsed time for package gmp:x86-windows: 2.367 min
```

**Comment by LegalizeAdulthood** [comment]
`vcpkg install cgal:x64-windows` also builds gmp without difficulty.

**Comment by PhoebeHui** [comment]
This issue will be fixed by PR https://github.com/microsoft/vcpkg/pull/16478

**Comment by lrineau** [comment]
> This issue will be fixed by PR #16478

Should this issue be closed, now?
